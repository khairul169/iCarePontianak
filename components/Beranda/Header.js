import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { Icon, SearchBar, StatusBar } from "../index";
import { statusBarHeight } from "../StatusBar";

const HomeHeader = () => {
  const headerStyle = [
    styles.header,
    {
      marginTop: statusBarHeight
    }
  ];

  return (
    <View style={styles.headerContainer}>
      <StatusBar />

      <View style={headerStyle}>
        <Icon name="home-plus" size={24} color="#4f6780" />
        <Text style={styles.headerTitle}>iCare Pontianak</Text>
        <Icon
          type="Ionicons"
          name="ios-notifications-outline"
          size={20}
          color="#525252"
        />
      </View>

      <SearchBar marginTop={12} backgroundColor="#ECEFF1" />
    </View>
  );
};

const styles = StyleSheet.create({
  headerContainer: {
    padding: 16,
    backgroundColor: "#fff"
  },
  header: {
    flexDirection: "row",
    alignItems: "center"
  },
  headerAppIcon: {
    width: 28,
    height: 28,
    tintColor: "#4f6780"
  },
  headerTitle: {
    color: "#333",
    fontSize: 16,
    fontWeight: "100",
    marginLeft: 16,
    flex: 1
  }
});

export default HomeHeader;
