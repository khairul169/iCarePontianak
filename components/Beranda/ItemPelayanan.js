import React from "react";
import { View, Text, StyleSheet, Image, TouchableOpacity } from "react-native";
import PropTypes from "prop-types";

const ItemPelayanan = ({ title, image, onPress, border }) => {
  return (
    <View style={[styles.container, border && styles.border]}>
      <TouchableOpacity style={styles.item} onPress={onPress}>
        <Image source={image} style={styles.pelayananImage} />
        <Text style={styles.pelayananTitle}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

ItemPelayanan.propTypes = {
  title: PropTypes.string,
  image: PropTypes.any,
  onPress: PropTypes.func,
  border: PropTypes.bool
};

ItemPelayanan.defaultProps = {
  border: true
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexBasis: "30%"
  },
  border: {
    borderBottomColor: "#CFD8DC",
    borderRightColor: "#CFD8DC",
    borderBottomWidth: 1,
    borderRightWidth: 1
  },
  item: {
    flex: 1,
    alignItems: "center",
    padding: 16
  },
  pelayananImage: {
    flex: 1,
    resizeMode: "contain"
  },
  pelayananTitle: {
    color: "#525252",
    fontSize: 11,
    marginTop: 12,
    textAlign: "center"
  }
});

export default ItemPelayanan;
