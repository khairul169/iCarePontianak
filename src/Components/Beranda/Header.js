import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  StatusBar,
  Platform
} from "react-native";
import Icon from "../Icon";

const lightStatusBar = Platform.OS !== "android" || Platform.Version > 22;

const HomeHeader = () => {
  return (
    <View style={styles.headerContainer}>
      <StatusBar
        barStyle={lightStatusBar ? "dark-content" : "light-content"}
        backgroundColor={lightStatusBar ? "#fff" : "#454545"}
      />

      <View style={styles.header}>
        <Icon name="home-plus" size={24} color="#4f6780" />
        <Text style={styles.headerTitle}>iCare Pontianak</Text>
        <Icon
          type="Ionicons"
          name="ios-notifications-outline"
          size={20}
          color="#525252"
        />
      </View>

      <View style={styles.searchBox}>
        <TextInput
          style={styles.searchInput}
          placeholder="Mau cari tindakan atau bantuan apa?"
        />
        <Icon
          type="Ionicons"
          name="md-search"
          size={18}
          color="#4f6780"
          style={styles.searchIcon}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  headerContainer: {
    padding: 16,
    backgroundColor: "#fff",
    elevation: 3
  },
  header: {
    flexDirection: "row",
    alignItems: "center"
  },
  headerAppIcon: {
    width: 28,
    height: 28,
    tintColor: "#4f6780"
  },
  headerTitle: {
    color: "#333",
    fontSize: 16,
    fontWeight: "100",
    marginLeft: 16,
    flex: 1
  },
  searchBox: {
    marginTop: 12,
    backgroundColor: "#eee",
    borderRadius: 5,
    flexDirection: "row",
    alignItems: "center"
  },
  searchInput: {
    paddingHorizontal: 10,
    paddingVertical: 4,
    flex: 1
  },
  searchIcon: {
    marginHorizontal: 10
  }
});

export default HomeHeader;
