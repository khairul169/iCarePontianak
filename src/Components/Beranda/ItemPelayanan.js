import React from "react";
import { Text, StyleSheet, Image, TouchableOpacity } from "react-native";
import PropTypes from "prop-types";

const ItemPelayanan = ({ title, image, height, onPress }) => {
  const containerStyle = [styles.container, { height }];
  return (
    <TouchableOpacity style={containerStyle} onPress={onPress}>
      <Image source={image} style={styles.pelayananImage} />
      <Text style={styles.pelayananTitle}>{title}</Text>
    </TouchableOpacity>
  );
};

ItemPelayanan.propTypes = {
  title: PropTypes.string,
  image: PropTypes.any,
  height: PropTypes.number,
  onPress: PropTypes.func
};

ItemPelayanan.defaultProps = {
  height: 80
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    minWidth: "25%",
    padding: 12
  },
  pelayananImage: {
    flex: 1,
    height: "100%",
    resizeMode: "contain"
  },
  pelayananTitle: {
    color: "#525252",
    fontSize: 11,
    marginTop: 8,
    textAlign: "center"
  }
});

export default ItemPelayanan;
