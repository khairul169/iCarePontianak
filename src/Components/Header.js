import React from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import PropTypes from "prop-types";
import StatusBar, { statusBarHeight } from "./StatusBar";
import Icon from "./Icon";

export const HeaderItem = ({ style, children, onPress }) => {
  if (onPress) {
    return (
      <TouchableOpacity
        style={[styles.headerItemContainer, style]}
        onPress={onPress}
      >
        {children}
      </TouchableOpacity>
    );
  }

  return <View style={styles.headerItemContainer}>{children}</View>;
};

HeaderItem.propTypes = {
  onPress: PropTypes.func
};

export const HeaderIcon = ({ type, name, onPress, right }) => {
  return (
    <HeaderItem style={right && styles.headerItemRight} onPress={onPress}>
      <Icon type={type} name={name} size={22} color="#333" />
    </HeaderItem>
  );
};

HeaderIcon.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  right: PropTypes.bool
};

const Header = ({
  title,
  navigation,
  left,
  right,
  backButton,
  transparent
}) => {
  const goBack = () => {
    navigation && navigation.goBack();
  };

  const containerStyle = [
    styles.header,
    transparent && styles.headerTransparent,
    { paddingTop: statusBarHeight }
  ];

  return (
    <View style={containerStyle}>
      <StatusBar />

      <View style={styles.headerItem}>
        {backButton && <HeaderIcon name="arrow-left" onPress={goBack} />}
        {left}
      </View>

      <Text style={styles.headerTitle}>{title && title.toUpperCase()}</Text>

      <View style={[styles.headerItem, styles.right]}>{right}</View>
    </View>
  );
};

Header.propTypes = {
  title: PropTypes.string,
  transparent: PropTypes.bool,
  left: PropTypes.element,
  right: PropTypes.element,
  backButton: PropTypes.bool,
  navigation: PropTypes.object
};

Header.defaultProps = {
  backButton: false,
  transparent: false
};

const styles = StyleSheet.create({
  header: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 16,
    height: 80,
    backgroundColor: "#fff"
  },
  headerTransparent: {
    backgroundColor: "rgba(0, 0, 0, 0)",
    position: "absolute",
    left: 0,
    top: 0,
    right: 0,
    zIndex: 1
  },
  headerItem: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center"
  },
  headerItemContainer: {
    flex: 1,
    alignSelf: "stretch",
    alignItems: "flex-start",
    justifyContent: "center"
  },
  headerItemRight: {
    alignItems: "flex-end"
  },
  headerTitle: {
    flex: 3,
    fontSize: 16,
    fontWeight: "bold",
    color: "#424242",
    textAlign: "center"
  }
});

export default Header;
