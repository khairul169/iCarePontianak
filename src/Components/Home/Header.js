import React from "react";
import { View, Text, StyleSheet, TextInput, StatusBar } from "react-native";
import Icon from "../Icon";

const HomeHeader = () => {
  return (
    <View style={styles.headerContainer}>
      <StatusBar barStyle="dark-content" backgroundColor="#fff" />

      <View style={styles.header}>
        <Icon name="home-plus" size={24} color="#4f6780" />
        <Text style={styles.headerTitle}>iCare Pontianak</Text>
        <Icon
          type="Ionicons"
          name="ios-notifications-outline"
          size={20}
          color="#525252"
        />
      </View>

      <View style={styles.searchBox}>
        <TextInput
          style={styles.searchInput}
          placeholder="Mau cari tindakan atau bantuan apa?"
        />
        <Icon
          type="Ionicons"
          name="md-search"
          size={18}
          color="#4f6780"
          style={styles.searchIcon}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  headerContainer: {
    padding: 16,
    backgroundColor: "#fff"
  },
  header: {
    flexDirection: "row",
    alignItems: "center"
  },
  headerAppIcon: {
    width: 28,
    height: 28,
    tintColor: "#4f6780"
  },
  headerTitle: {
    color: "#333",
    fontSize: 16,
    fontWeight: "100",
    marginLeft: 16,
    flex: 1
  },
  searchBox: {
    marginTop: 12,
    backgroundColor: "#eee",
    borderRadius: 5,
    flexDirection: "row",
    alignItems: "center"
  },
  searchInput: {
    paddingHorizontal: 10,
    paddingVertical: 4,
    flex: 1
  },
  searchIcon: {
    marginHorizontal: 10
  }
});

export default HomeHeader;
