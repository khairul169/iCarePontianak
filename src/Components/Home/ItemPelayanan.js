import React from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import PropTypes from "prop-types";

const ItemPelayanan = ({ title, image, borderLeft }) => {
  return (
    <View
      style={[styles.pelayananItem, borderLeft && styles.pelayananitemBordered]}
    >
      <Image source={image} style={styles.pelayananImage} />
      <Text style={styles.pelayananTitle}>{title}</Text>
    </View>
  );
};

ItemPelayanan.propTypes = {
  title: PropTypes.string,
  image: PropTypes.any,
  borderLeft: PropTypes.bool
};

const styles = StyleSheet.create({
  pelayananItem: {
    flex: 1,
    alignItems: "center",
    minHeight: 52,
    minWidth: "30%"
  },
  pelayananitemBordered: {
    borderLeftColor: "#eee",
    borderLeftWidth: 1
  },
  pelayananImage: {
    flex: 1,
    height: "100%",
    resizeMode: "contain"
  },
  pelayananTitle: {
    color: "#525252",
    fontSize: 12,
    marginTop: 8
  }
});

export default ItemPelayanan;
