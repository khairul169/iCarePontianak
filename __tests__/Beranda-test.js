import React from "react";
import renderer from "react-test-renderer";

import Header from "../components/Beranda/Header";
import ItemPelayanan from "../components/Beranda/ItemPelayanan";

describe("<Header />", () => {
  const defaultProps = {};
  const wrapper = renderer.create(<Header {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<ItemPelayanan />", () => {
  const defaultProps = {
    title: "Item"
  };
  const wrapper = renderer.create(<ItemPelayanan {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
