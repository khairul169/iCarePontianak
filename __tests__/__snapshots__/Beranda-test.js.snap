// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 16,
    }
  }
>
  <View />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "marginTop": 20,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#4f6780",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      若
    </Text>
    <Text
      style={
        Object {
          "color": "#333",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "100",
          "marginLeft": 16,
        }
      }
    >
      iCare Pontianak
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#525252",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#ECEFF1",
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginTop": 12,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Mau cari tindakan atau bantuan apa?"
      placeholderTextColor="#78909C"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#333",
            "flex": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 4,
          },
          Object {
            "color": "#37474F",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#37474F",
            "fontSize": 18,
          },
          Object {
            "marginHorizontal": 10,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`<ItemPelayanan /> render 1`] = `
<View
  accessible={true}
  clickable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 80,
      "minWidth": "25%",
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <Image
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "resizeMode": "contain",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#525252",
        "fontSize": 11,
        "marginTop": 8,
        "textAlign": "center",
      }
    }
  >
    Item
  </Text>
</View>
`;
